test/swift_class_test.exs

```diff
@@ -113,10 +119,42 @@ defmodule SwiftClassTest do
     end
 
     test "parses attr value references" do
-      input = "foo(attr(bar))"
-      output = [["foo", [["bar", :attr]], nil]]
+      input = "foo(attr(\"bar\"))"
+      output = [["foo", [["Attr", "bar"]], nil]]
 
       assert parse(input) == output
     end
   end
+
+  describe "class block parser" do
+    input = """
+    "color-" <> color_name do
+      foo(true)
+      color(color_name)
+      bar(false)
+    end
+
+    "color-red" do
+      color(:red)
+    end
```

Comment:
@bcardarella, these really look like things that should be implemented with a DSL. Have you considered that approach instead of trying to do parsing?

Reply:
@NduatiK Elixr won't parse this the way you think it will. We'll need to parse with NimbleParsec and produce the AST that I can then use to generate function signatures

Reply:
the generated code isn't going to be directly accessible as a public API, it's intended for stylesheet generation. We'll have control over how the functions are called